.github/workflows/test.yml

```diff
@@ -52,9 +52,9 @@ jobs:
       id: go
 
     - name: Verify
-      uses: golangci/golangci-lint-action@v4
+      uses: golangci/golangci-lint-action@v6
```

Comment:
sooo in CI:
```
run golangci-lint
  Running [/home/runner/golangci-lint-1.56.1-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=go-controller --modules-download-mode=vendor --timeout=15m0s --verbose] in [/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller] ...
  level=info msg="[config_reader] Config search paths: [./ /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller /home/runner/work/ovn-kubernetes/ovn-kubernetes /home/runner/work/ovn-kubernetes /home/runner/work /home/runner /home /]"
  level=info msg="[config_reader] Used config file .golangci.yml"
```
when a PR is submitted we are gonna use the file to which we added all the linter improvements so that's good cause it will show up red if some code doesn't comply.